plugins/woocommerce/src/Admin/API/Reports/Products/DataStore.php

```diff
@@ -432,17 +466,144 @@ public static function sync_order_products( $order_id ) {
+					}
+					return $carry;
+				},
+				array(
+					'variation_ids' => array(),
+					'product_ids'   => array(),
+				)
+			);
+
+			// A helper function to determine if the order item should be included in the report.
+			// It's only used in the below array_filter function.
+			$should_include_order_items = function ( $partial_refund_product_revenue, $order_item, $id, $id_key = 'product_ids' ) {
+				// If the product is not partially refunded, include it.
+				if ( ! isset( $partial_refund_product_revenue[ $id_key ][ $id ] ) ) {
+					return true;
+				}
+
+				$net_revenue = round( $order_item->get_total( 'edit' ), $decimals );
+
+				// If the product is not 100% partially refunded, include it.
+				// So we can store the difference in the product lookup table later.
+				if ( abs( $partial_refund_product_revenue[ $id_key ][ $id ] ) !== abs( $net_revenue ) ) {
+					return true;
+				}
+
+				return false;
+			};
+
+			// Filter the items that is already being patially refunded from the parent order item.
+			$order_items = array_filter(
+				$parent_order_items,
+				function ( $order_item ) use ( $partial_refund_product_revenue, $should_include_order_items ) {
+					$variation_id = $order_item->get_variation_id();
+
+					if ( $variation_id ) {
+						return $should_include_order_items( $partial_refund_product_revenue, $order_item, $variation_id, 'variation_ids' );
+					}
+
+					return $should_include_order_items( $partial_refund_product_revenue, $order_item, $order_item->get_product_id() );
+				}
+			);
+		}
+
 		foreach ( $order_items as $order_item ) {
 			$order_item_id = $order_item->get_id();
 			unset( $existing_items[ $order_item_id ] );
 			$product_qty         = $order_item->get_quantity( 'edit' );
+			$product_id          = $order_item->get_product_id( 'edit' );
+			$variation_id        = $order_item->get_variation_id( 'edit' );
 			$shipping_amount     = $order->get_item_shipping_amount( $order_item );
 			$shipping_tax_amount = $order->get_item_shipping_tax_amount( $order_item );
 			$coupon_amount       = $order->get_item_coupon_amount( $order_item );
+			$net_revenue         = round( $order_item->get_total( 'edit' ), $decimals );
+
+			// If the order is a full refund and there is no order items. The order item here is the parent order item.
+			if ( $is_full_refund_without_line_items ) {
+				$key = $variation_id ? 'variation_ids' : 'product_ids';
+				$id  = $variation_id ? $variation_id : $product_id;
+				if ( isset( $partial_refund_product_revenue[ $key ][ $id ] ) ) {
+					// If a single line item was refunded 60% then fully refunded after, we need store the difference in the product lookup table.
+					// E.g. A product costs $100, it was previously partially refunded $60, then fully refunded $40.
+					// So it will be -abs( 100 + (-60) ) = -40.
+					$net_revenue = -abs( $net_revenue + $partial_refund_product_revenue[ $key ][ $id ] );
+				} else {
+					$net_revenue = -abs( $net_revenue );
+				}
```

Comment:
If the previous two suggestions to simplify are taken into account then we can reduce this code a bit more.

```suggestion
			if ( $is_full_refund_without_line_items ) {
				$id             = $variation_id ?: $product_id;
				$partial_refund = $partial_refund_product_revenue[ $id ] ?? 0;
				$net_revenue    = -abs( $net_revenue + $partial_refund );
				
				// Skip items that have already been fully refunded (single or multiple partial refunds).
				if ( 0.0 === $net_revenue ) {
					continue;
				}
```